pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -267,29 +256,17 @@ func (manager *gceServiceManager) GetRegionalDiskFromCloudProvider(
 
 func (manager *gceServiceManager) DeleteDiskOnCloudProvider(
 	zone string,
-	diskName string) (gceObject, error) {
-	return manager.gce.service.Disks.Delete(
-		manager.gce.projectID, zone, diskName).Do()
+	diskName string) error {
+	return manager.gce.c.Disks().Delete(context.Background(), meta.ZonalKey(diskName, zone))
 }
 
 func (manager *gceServiceManager) DeleteRegionalDiskOnCloudProvider(
-	diskName string) (gceObject, error) {
+	diskName string) error {
 	if utilfeature.DefaultFeatureGate.Enabled(features.GCERegionalPersistentDisk) {
-		return manager.gce.serviceBeta.RegionDisks.Delete(
-			manager.gce.projectID, manager.gce.region, diskName).Do()
+		return manager.gce.c.BetaRegionDisks().Delete(context.Background(), meta.RegionalKey(diskName, manager.gce.region))
 	}
 
-	return nil, fmt.Errorf("the regional PD feature is only available with the %s Kubernetes feature gate enabled", features.GCERegionalPersistentDisk)
-}
-
-func (manager *gceServiceManager) WaitForZoneOp(
-	op gceObject, zone string, mc *metricContext) error {
-	return manager.gce.waitForZoneOp(op, zone, mc)
-}
-
-func (manager *gceServiceManager) WaitForRegionalOp(
-	op gceObject, mc *metricContext) error {
-	return manager.gce.waitForRegionOp(op, manager.gce.region, mc)
+	return fmt.Errorf("the regional PD feature is only available with the %s Kubernetes feature gate enabled", features.GCERegionalPersistentDisk)
```

Comment:
Nit: I would flip the conditions so you return the error in the conditional and move the primary action outside of the block.

Reply:
Will keep it as is - I want to make clear exactly what code block gets disabled if the feature gate is turned off. For other conditionals yeah, I agree

Reply:
Keeping it is fine. I actually feel the opposite - I rather see the rest of the function as having passed this feature gate. If the enabled case was more than one line, I'd feel stronger about it.

Reply:
Updated